lib/constants.js

```diff
@@ -20,6 +20,7 @@ module.exports = {
 , MAC_ADDR_CHECK: /^([0-9a-fA-F][0-9a-fA-F]:){5}([0-9a-fA-F][0-9a-fA-F])$/
 , MAX_REQUEST_TIMEOUT: 300000
 , MS_IN_A_DAY: 86400000
+, NON_PRINTABLE_CHAR_RE: /[^\x20-\x7E]/g
```

Comment:
It's funny we added the same logic on the rust agent recently... I was getting host names and versions with line breaks included: https://github.com/logdna/logdna-agent-v2/blob/master/common/config/src/lib.rs#L154

I thought it was related to the rust library but it seems that it's an OS thing, so we should probably add it across the board on all client libraries (ruby, python, ...).

Reply:
Agreed.  We can also fix the agents from sending the newline, but having the protection here is best.